typespec-extension/src/code-model-builder.ts

```diff
@@ -1975,164 +1887,91 @@ export class CodeModelBuilder {
         },
       },
     });
-    schema.crossLanguageDefinitionId = getCrossLanguageDefinitionId(type);
+    schema.crossLanguageDefinitionId = type.crossLanguageDefinitionId;
     return this.codeModel.schemas.add(schema);
   }
 
-  private processConstantSchemaForLiteral(
-    type: StringLiteral | NumericLiteral | BooleanLiteral,
-    name: string,
-  ): ConstantSchema {
-    const valueType =
-      type.kind === "String"
-        ? this.stringSchema
-        : type.kind === "Boolean"
-          ? this.booleanSchema
-          : isAllValueInteger([type.value])
-            ? this.integerSchema
-            : this.doubleSchema;
+  private processConstantSchemaFromSdkType(type: SdkConstantType, name: string): ConstantSchema {
+    const valueType = this.processSchemaFromSdkType(type.valueType, type.valueType.kind);
 
     return this.codeModel.schemas.add(
-      new ConstantSchema(name, this.getDoc(type), {
-        summary: this.getSummary(type),
+      new ConstantSchema(name, this.getDoc(type.__raw), {
+        summary: this.getSummary(type.__raw),
         valueType: valueType,
         value: new ConstantValue(type.value),
       }),
     );
   }
 
-  private processConstantSchemaForEnumMember(type: EnumMember, name: string): ConstantSchema {
-    const valueType = this.processSchema(type.enum, this.getName(type.enum));
+  private processConstantSchemaFromEnumValueFromSdkType(type: SdkEnumValueType, name: string): ConstantSchema {
+    const valueType = this.processSchemaFromSdkType(type.enumType, type.enumType.name);
 
     return this.codeModel.schemas.add(
-      new ConstantSchema(name, this.getDoc(type), {
-        summary: this.getSummary(type),
+      new ConstantSchema(name, this.getDoc(type.__raw), {
+        summary: this.getSummary(type.__raw),
         valueType: valueType,
         value: new ConstantValue(type.value ?? type.name),
       }),
     );
   }
 
-  private processConstantSchemaForUnionVariant(type: UnionVariant, name: string): ConstantSchema {
```

Comment:
Which TCGC type is replacing this?

Reply:
The logic is to process constant union variant, so in [`processSchemaFromSdkType`](https://github.com/Azure/autorest.java/blob/647a0208d1fb716885b36b3224aa9f574a4dc3ad/typespec-extension/src/code-model-builder.ts#L2190), the sdktype kind is `"constant"`. It will goes into this [logic](https://github.com/Azure/autorest.java/blob/647a0208d1fb716885b36b3224aa9f574a4dc3ad/typespec-extension/src/code-model-builder.ts#L1715).

Reply:
Are you sure?

I thought it may be included in `enumvalue` branch, but unsure.

Basically the difference is
```
model {
  prop: Enum.Value1;
}
```
and whether that `Enum` is `enum`, or `union`.

Reply:
I thought you are asking `processConstantSchemaForUnionVariant` which processes constant union variant type. So I'm referring to below case, when processing "a", it is constant type. 
```ts
model A {
   p: "a" | "b" | string
}
```


For enum value as property type. Take below test case as example:
```

model Operation {
  name: "Read" | "Write";
  best: true;
  age: 50;
  priority: Priority;
  color: ColorModel;
  unit: Unit;
  priorityValue: Priority.Low;
  colorValue: Color.Green;
  colorModelValue: ColorModel.Blue;
  unitValue: Unit.Milligram;
}
```

TCGC response for `Priority.Low` is 
![image](https://github.com/Azure/autorest.java/assets/87355844/1e3ff775-fe7e-4c5b-a12c-f169f809d92e)

Reply:
Yes, I AM asking `processConstantSchemaForUnionVariant`, which is for this `ColorModel.Blue`.

Reply:
The `a` in `"a" | "b" | string` is not constant type either, it would belong to the Enum process.

Reply:
Yes, you are right, the example is wrong. should be: `"a" | "b" | int32`. And "a" is constant union variant.

ok, I get your point, you are saying `ColorModel` is union. For `colorModelValue: ColorModel.Blue;` field, The response is like below:
![image](https://github.com/Azure/autorest.java/assets/87355844/0059132b-ff88-4c46-9b81-90e9a17d477a)

It is also 'enumvalue' kind. And `ColorModel` is  'enum' kind.

Reply:
`"a" | "b" | int32` won't go here either. It would go to the processUnion.

The "a" in the Union/Enum along is never a constant (union variant or not).